Configurations/50-win-arm.conf

```diff
@@ -0,0 +1,24 @@
+# Windows on ARM targets. ARM compilers are additional components in
+# VS2017, i.e. they are not installed by default. When installed,
+# there are no "ARM Tool Command Prompt"s on Start menu, you have
+# to locate vcvarsall.bar and act accordingly. VC-WIN32-ARM has
+# received limited testing testing with evp_test.exe on Windown 10
+# IoT Core, but not VC-WIN64-ARM, no hardware... In other words they
+# are not actually supported...
+
+my %targets = (
+    "VC-WIN32-ARM" => {
+        inherit_from    => [ "VC-noCE-common" ],
+        defines         => add("_ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE"),
+        bn_ops          => "BN_LLONG RC4_CHAR EXPORT_VAR_AS_FN",
+        lflags          => add("/NODEFAULTLIB:kernel32.lib"),
+        ex_libs         => "onecore.lib onecore_downlevel.lib",
+    },
+    "VC-WIN64-ARM" => {
+        inherit_from    => [ "VC-noCE-common" ],
+        defines         => add("_ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE"),
+        bn_ops          => "SIXTY_FOUR_BIT RC4_CHAR EXPORT_VAR_AS_FN",
+        lflags          => add("/NODEFAULTLIB:kernel32.lib"),
+        ex_libs         => "onecore.lib onecore_downlevel.lib",
+    },
```

Comment:
So, errr, where is `VC-WIN32-ARM` supposed to be installed?  I notice that you're not adding any flavor, so it will end up being installed like a VC-W64 flavor...

Reply:
Well, define "installation" in cross-compilation scenario. In *pure* cross-compilation scenario when target file system has *nothing* to do with host's. Nor has it any of host's facilities. [E.g. what meaning do headers and lib files have to target system when it doesn't and never will have compiler?] Easiest to imagine on CE's example. There is no C: on CE device, or `Program Files`. So why one would compile host's values into target's binary code? Answer is that you pretty much have no other choice but to specify **target's** path explicitly with --prefix [and the other one], then specify DESTDIR at nmake install time at host, and finally do whatever undefined magic it takes to transfer executable code to the target system. [Note that you might as well choose to skip nmake install.] But let's consider that you didn't specify --prefix in this win-arm case. There actually is `C:\Program Files` [at least] on my device. So it's actually not inappropriate **default** for this target installation. In fact, if you think about it, it wouldn't be inappropriate choice for any non-dual-ABI non-CE system. "Non-dual-ABI" means that you can execute only one kind of binaries. In other words having "VC-W64" install path is actually meaningful default.

As for "VC-W64" flavour in more general sense. With last sentence in mind one can argue that it's more appropriate to actually view it as "VC-non-dual-ABI" or ... "VC-common" ;-) Also note that all that "VC-W32" dance is rather about dual-ABI than anything else. So imagine we rename it to "VC-WOW" instead. Wouldn't it be helpful? Just in case "WOW" stands for "Windows on Windows" and is kind of established term.

[Just in case for reference. Is it hard to get hold of Windows ARM device now? Not really, there are images for Raspberry Pi. Though I myself don't run it on Raspberry Pi.]

Reply:
It's not inappropriate to add remark about host vs. target file systems into commentary section, so I'll do that...

Reply:
> rename it [VC-W32] to "VC-WOW"

Shall we?

Reply:
I see no problem with VC-WOW per se.